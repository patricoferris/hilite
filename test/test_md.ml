let md =
  {| OCaml is a great language, e.g.

```ocaml
let add a b = a + b
```

And some bash...

```bash
~/$ ls ~/.opam
```

And some sh...

```sh
~/$ ls ~/.opam
```
|}

let test_transform () =
  let omd_t = Cmarkit.Doc.of_string md |> Hilite.Md.transform in
  Format.printf "%s" (Cmarkit_html.of_doc ~safe:false omd_t)

let () = test_transform ()
