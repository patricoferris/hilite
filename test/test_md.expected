<p>OCaml is a great language, e.g.</p>
<pre><code><span class='ocaml-keyword'>let</span><span class='ocaml-source'> </span><span class='ocaml-entity-name-function-binding'>add</span><span class='ocaml-source'> </span><span class='ocaml-source'>a</span><span class='ocaml-source'> </span><span class='ocaml-source'>b</span><span class='ocaml-source'> </span><span class='ocaml-keyword-operator'>=</span><span class='ocaml-source'> </span><span class='ocaml-source'>a</span><span class='ocaml-source'> </span><span class='ocaml-keyword-operator'>+</span><span class='ocaml-source'> </span><span class='ocaml-source'>b</span><span class='ocaml-source'>
</span></code></pre>
<p>And some bash...</p>
<pre><code><span class='bash-keyword-operator-tilde'>~</span><span class='bash-source'>/$ ls </span><span class='bash-keyword-operator-tilde'>~</span><span class='bash-source'>/.opam
</span></code></pre>
<p>And some sh...</p>
<pre><code><span class='sh-keyword-operator-tilde'>~</span><span class='sh-source'>/$ ls </span><span class='sh-keyword-operator-tilde'>~</span><span class='sh-source'>/.opam
</span></code></pre>
